core/codegen_next/src/prefixing_vec.rs

```diff
@@ -0,0 +1,32 @@
+use proc_macro::TokenStream;
+use proc_macro2::TokenStream as TokenStream2;
+
+use syn::Path;
+
+use parser::{Parser, Seperator, Result as PResult};
+use utils::IdentExt;
+
+#[inline]
+fn prefix_path(prefix: &str, path: &mut Path) {
+    let mut last_seg = path.segments.last_mut().expect("last path segment");
+    let last_value = last_seg.value_mut();
+    last_value.ident = last_value.ident.prepend(prefix);
+}
+
+pub fn prefixing_vec_macro<F>(prefix: &str, to_expr: F, args: TokenStream) -> PResult<TokenStream>
+        where F: FnMut(Path) -> TokenStream2
+{
+    let mut parser = Parser::new(args);
+    let mut paths = parser.parse_sep(Seperator::Comma, |p| {
+        p.parse::<Path>()
+    })?;
+    parser.eof()?;
```

Comment:
Testing for `eof()` is necessary, otherwise something like `routes![a, b c]` will be accepted and parse as only `a,b`. The compiler plugin macro says it expected `,`, `::`, or `<eof>`. This proc_macro currently just says `expected eof`.

Reply:
I will try replacing the error message with a better one, like "expected paths separated by commas".